packages/aws-cdk-lib/aws-docdb/lib/cluster.ts

```diff
@@ -495,6 +521,19 @@ export class DatabaseCluster extends DatabaseClusterBase {
       throw new Error('KMS key supplied but storageEncrypted is false');
     }
 
+    const validEngineVersionRegex = /^(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)$/;
+    if (props.engineVersion !== undefined && !validEngineVersionRegex.test(props.engineVersion)) {
+      throw new Error(`Invalid engine version: ${props.engineVersion}. Engine version must be in the format x.y.z`);
+    }
+
+    if (
+      props.storageType === StorageType.IOPT1
+      && props.engineVersion !== undefined
+      && Number(props.engineVersion.split('.')[0]) < MIN_ENGINE_VERSION_FOR_IO_OPTIMIZED_STORAGE
```

Comment:
The I/O optimized storage type can only be specified from engine version 5.0.0 onwards. 

It would have been better if `engineVersion` were defined as an enum, but unfortunately, it is a string type, so I am validating it in this manner.

I look forward to hearing if there are any better methods.

Reply:
Can you please add the documentation link to this? I believe this is the only official source mentioning the version that this feature became available in: https://docs.aws.amazon.com/documentdb/latest/developerguide/release-notes.html#release-notes.11-21-2023

Reply:
Can you also update the docstring for the `engineVersion` prop to clarify that the default engine version is the latest major version of doc db (no need to indicate the actual number, just that it's the latest major version)? It currently reads "the default engine version" which is not helpful.

Reply:
I've added the documentation link and updated default description of `engineVersion`.